### PR TITLE
Re-enable tables functional tests

### PR DESCRIPTION
**Pull request by pakrym**
I kept the original grouping but simplified the invocation pattern to avoid using the Queue trigger.

I also removed the mock storage implementation and made tests fully live.

Removed test shared code that's not used anymore.

### CODE REVIEW

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/Tables/TablesTestEnvironment.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.TestFramework;
+
+namespace Microsoft.Azure.WebJobs.Host.FunctionalTests
+{
+    public class TablesTestEnvironment: TestEnvironment
```
Comment (JoshLove-msft) [inline sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/Tables/TablesTestEnvironment.cs:8]: Doesn't this already exist from Tables test project?

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/WebJobsShared/TestHelpers.cs [hunk 2]
```diff
@@ -145,148 +21,13 @@ public static IHostBuilder ConfigureDefaultTestHost(this IHostBuilder builder, A
...
-                        services.AddSingleton<INameResolver>(nameResolver);
-                    }
-                    if (activator != null)
-                    {
-                        services.AddSingleton<IJobActivator>(activator);
-                    }
-                });
-        }
+
         public static IHostBuilder ConfigureTestLogger(this IHostBuilder builder)
```
Comment (JoshLove-msft) [inline sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/WebJobsShared/TestHelpers.cs:25]: Don't we already have this in shared source?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/WebJobsShared/TestHelpers.cs (hunk 2 above)
Comment (pakrym) [reply sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/WebJobsShared/TestHelpers.cs:25]: Yep, I'll switch to it in the next PR.

File: sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/Tables/TablesTestEnvironment.cs (hunk 1 above)
Comment (pakrym) [reply sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/Tables/TablesTestEnvironment.cs:8]: It does but I'm not sure I want to source include it. Wanted to keep projects decoupled.

**Review by christothes — APPROVED** [review]

**Comment by pakrym** [comment]
^ unrelated flakiness

**Pull request merged**